src/helpers/DiffHelpers.ts

```diff
@@ -0,0 +1,104 @@
+import {
+  Operation,
+  ReplaceOperation,
+  AddOperation,
+  getValueByPointer,
+} from 'fast-json-patch';
+
+export interface DiffOutput {
+  action: string;
+  path: string;
+  index?: number;
+  before?: any;
+  after?: any;
+}
+
+type ValueOperation = ReplaceOperation<any> | AddOperation<any>;
+
+/**
+ * Formats the action to have one of `add`, `remove` or `edit` values
+ * @param {String} action The action performed by the diff library
+ * @returns {String} The formatted action output
+ */
+export function formatAction(action: string): string {
+  if (action === 'replace') {
+    return 'edit';
+  }
+  // since `add` and `remove` are already provided by the library, we don't need to change that.
+  return action;
+}
+
+/**
+ * Get the `before` value from the first document
+ * @param firstDocument The first document which we will get the before value from
+ * @param {String} path The path from where we can get the value
+ * @returns The value from the first document
+ */
+export function getBeforeValue(firstDocument: any, path: string): any {
+  // since our current diffing library doesn't provide the `before` value
+  // we have to get it manually from the first document
+  return getValueByPointer(firstDocument, path);
+}
+
+/**
+ * Get the `after` value
+ * @param diffObject A single diff object
+ * @returns The value that is present inside the second document
+ */
+export function getAfterValue(diffObject: Operation): any {
+  // The diffing library already provides the `after` value for `add` or `edit` actions
+  return (diffObject as ValueOperation).value;
+}
+
+/**
+ * Sets the index property in case of an array change
+ * @param changeObject The change object
+ * @param {String} path The original path
+ */
+export function setIndex(
+  changeObject: DiffOutput,
+  firstDocument: any,
+  path: string
+): void {
+  const splittedPath = path.split('/');
+  const lastPathElement = splittedPath[splittedPath.length - 1];
+  const lastElementNumber = Number(lastPathElement);
+  splittedPath.pop();
+  const assembledPath = splittedPath.join('/');
+  if (
+    !Number.isNaN(lastElementNumber) &&
+    Array.isArray(getValueByPointer(firstDocument, assembledPath))
+  ) {
+    // if the last element is a Number
+    // and the path before the last element is an array
+    // then the change has been made inside an array, therefore set its index
+    changeObject.index = lastElementNumber; // set the index
+  }
+}
```

Comment:
Sorry for that but I think that we should remove the `index` property from output item, but only indicate that the item is array, so `isArrayIndex: true` should be better. Why? At the moment we have duplicated the array's index value in the path and in `index` property, and it may not be understood by people.

Reply:
aah, you are right. Will add `isArrayIndex: true` then. Thanks :)